src/measure_theory/group/prod.lean

```diff
@@ -54,25 +54,23 @@ namespace measure_theory
 
 open measure
 
-/-- A shear mapping preserves the measure `μ.prod ν`.
+/-- The shear mapping `(x, y) ↦ (x, xy)` preserves the measure `μ × ν`.
 This condition is part of the definition of a measurable group in [Halmos, §59].
 There, the map in this lemma is called `S`. -/
-@[to_additive map_prod_add_eq /-" An additive shear mapping preserves the measure `μ.prod ν`. "-/]
-lemma map_prod_mul_eq [is_mul_left_invariant ν] :
-  map (λ z : G × G, (z.1, z.1 * z.2)) (μ.prod ν) = μ.prod ν :=
-((measure_preserving.id μ).skew_product measurable_mul
-  (filter.eventually_of_forall $ map_mul_left_eq_self ν)).map_eq
-
-/-- The function we are mapping along is `SR` in [Halmos, §59],
-  where `S` is the map in `map_prod_mul_eq` and `R` is `prod.swap`. -/
-@[to_additive map_prod_add_eq_swap /-"  "-/]
-lemma map_prod_mul_eq_swap [is_mul_left_invariant μ] :
-  map (λ z : G × G, (z.2, z.2 * z.1)) (μ.prod ν) = ν.prod μ :=
-begin
-  rw [← prod_swap],
-  simp_rw [map_map (measurable_snd.prod_mk (measurable_snd.mul measurable_fst)) measurable_swap],
-  exact map_prod_mul_eq ν μ
-end
+@[to_additive measure_preserving_prod_add
+  /-" An additive shear mapping `(x, y) ↦ (x, x + y)` preserves the measure `μ.prod ν`. "-/]
```

Comment:
```suggestion
  /-" The additive shear mapping `(x, y) ↦ (x, x + y)` preserves the measure `μ.prod ν`. "-/]
```

I'm also not sure if adding `additive` is useful here, since you give the expression and "shear" does usually refer to additive version too

Reply:
Thanks, I agree.